HIP/hip-100.md

```diff
@@ -0,0 +1,479 @@
+---
+hip: 100
+title: Signature and Wallet Providers
+author: Daniel Akhteorv <daniel@launchbadge.com>
+type: Standards Track
+category: Application
+needs-council-approval: Yes
+status: Draft
+created: 2022-01-20
+discussions-to: TBD
+updated: 2022-01-20
+---
+
+## **Abstract**
+
+Ethereum like signature and wallet providers with the Hedera SDKs
+
+## **Motivation**
+
+Cryptocurrency users are more used to Ethereum's signature and wallet providers instead
+of direclty hardcoding private keys into their projects. Supporting third party signature
+and wallet provider would be better for security and user experience.
+
+## **Specification**
+
+<details>
+<summary>abstract class Signer</summary>
+  
+### Methods
+
+##### `async signMessage` ( `messages`: `List` < `bytes` > ): `List` < `List` < `SignerSignature` > >
+
+Sign a list of messages
+
+**NOTE**: Each element in the outer list of the result is all the signatures for the
+at the same index.
+
+---
+
+##### `async signTransaction` ( `transaction`: `Transaction` )
+
+Signs the transaction
+
+**NOTE**: Use `Transaction.getSignatures()` to see the actual signatures.
+at the same index.
+
+---
+
+##### `async sendTransaction` ( `transaction`: `Transaction` ): `TransactionReceipt`
+
+Sign and send a transaction using the wallet
+
+**NOTE**: Unlike `Provider.sendTransaction()` this method will automatically wait for
+the receipt of the transaction.
+
+---
+
+##### `async checkTransaction` ( `transaction`: `Transaction` )
+
+Determines if all the properties requried are set and sets the transaction ID. If the transaction ID was already set it checks if the account ID of it is the same as the users.
+
+---
+
+##### `async populateTransaction` ( `transaction`: `Transaction` )
+
+Sets the transaction ID of the transaction to the current account ID of the signer.
+
```

Comment:
Suggest rename to `setTransactionId` for clarify. Although, the signer may also need to set the `node_account_id` to completely populate a transaction.
The SDK may not know the network the transaction is being sent to and therefore won't know the list of nodes